Sporadic failures in XrdClHttp

### Operating System

Ubuntu 24.04.3

### XRootD Version

5.9.1

### Description

We have seen the following errors in the CI

```
[2026-02-09 13:12:02.503245 +0000][Debug  ][XrdClHttp         ] Ignoring continuation of operation that has already completed
[2026-02-09 13:12:02.503256 +0000][Debug  ][File              ] [0xab8fc496f020@file://localhost/home/runner/work/xrootd/xrootd/build/tests/XRootD/http/test-hD4vpm/08.dat?oss.asize=8400430&xrdcl.requuid=6ed243f5-d4f9-4077-ba38-e9d9f8a193dd] Sending a write command for handle 0x5a to localhost
[2026-02-09 13:12:02.503263 +0000][Debug  ][XrdClHttp         ] Operation Failed?: false 
[2026-02-09 13:12:02.503267 +0000][Debug  ][XrdClHttp         ] curl operation at offset 0 failed with message: Operation previously failed; cannot continue
[2026-02-09 13:12:02.503342 +0000][Debug  ][Utility           ] Unable read 0 bytes at 0 from http://localhost:7094//home/runner/work/xrootd/xrootd/build/tests/XRootD/http/test-hD4vpm/08.ref: [ERROR] Internal error: Operation previously failed; cannot continue (read operation at offset 0)
```

The operation failed. Check part of the log line is from the following test commit: https://github.com/dynamic-entropy/xrootd/commit/2de6cf4d012386f35c2f9c3158240fe76248eac4 


As already mentioned in the code comments, this is a defence against an operation that has been requeued after completion.

However, one should not call `Fail()` or `Success()` on these ops, as they have already been marked so.

### Steps to Reproduce the Behavior

_No response_

### Additional Information

_No response_

--- a/src/XrdClHttp/XrdClHttpUtil.cc
+++ b/src/XrdClHttp/XrdClHttpUtil.cc
@@ -1107,7 +1107,6 @@ CurlWorker::Run() {
             // while the curl worker thread failed the transfer.
             if (op->IsDone()) {
                 m_logger->Debug(kLogXrdClHttp, "Ignoring continuation of operation that has already completed");
-                op->Fail(XrdCl::errInternal, 0, "Operation previously failed; cannot continue");
                 continue;
             }
             m_logger->Debug(kLogXrdClHttp, "Continuing the curl handle from op %p on thread %d", op.get(), getthreadid());

